### ISSUE TITLE
Too Many deleted fd

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmithdemos/leveldb#66]
```
I use leveldb on Linux, there are some read/write in each second.

After running three days, there are 1386 deleted fd. I think it is abnormal.

the attachment is the output of "ls -al /proc/$pid/fd"
```

---

Original issue reported on code.google.com by `yafei.zh...@langtaojin.com` on 12 Jan 2012 at 9:05

Attachments:
- [fd.log](https://storage.googleapis.com/google-code-attachments/leveldb/issue-66/comment-0/fd.log)

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I haven't seen this before.  If you have a LOG file from the db directory,
that could help shed light on the problem.

Some other possibilities: do you have a custom Env class that
is perhaps leaking file descriptors?  I am assuming that if there
was a general bug which was causing files to regularly not be
closed, somebody else would have reported it.

Since I don't know of a way to reproduce this, I won't be able
to help more.

```

---

Original comment by `san...@google.com` on 18 Jan 2012 at 6:05

**Comment by GoogleCodeExporter** [comment]
---

Original comment by `yafei.zh...@langtaojin.com` on 19 Jan 2012 at 2:46

Attachments:
- [LOG.tar.bz2](https://storage.googleapis.com/google-code-attachments/leveldb/issue-66/comment-3/LOG.tar.bz2)

**Comment by GoogleCodeExporter** [comment]
```
LOG.tar.bz2 is the LOG file.
I do not have a custom Env class.
I am trying to reproduce it and debug it now.
Thanks!
```

---

Original comment by `yafei.zh...@langtaojin.com` on 19 Jan 2012 at 2:48

**Comment by GoogleCodeExporter** [comment]
```
Thanks.  The LOG file you attached was helpful.  I have found the
problematic code and have a fix in progress.  You don't need to
do any further investigation.

```

---

Original comment by `san...@google.com` on 20 Jan 2012 at 5:11

**Comment by GoogleCodeExporter** [comment]
```
Fixed in change 3c8be108bfb5

```

---

Original comment by `san...@google.com` on 25 Jan 2012 at 11:13
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Thanks
```

---

Original comment by `yafei.zh...@langtaojin.com` on 31 Jan 2012 at 1:34
